src/NuGet.Services.AzureSearch/SearchService/AzureSearchTextBuilder.cs

```diff
@@ -47,90 +48,136 @@ private class AzureSearchQueryBuilder
             private readonly StringBuilder _result;
             private int _clauses;
 
-            public AzureSearchQueryBuilder()
+            public AzureSearchTextBuilder()
             {
                 _result = new StringBuilder();
                 _clauses = 0;
             }
 
             /// <summary>
-            /// Append unscoped search terms to the query. This can be called many times.
+            /// Append search terms to the query. These terms may match any field.
             /// </summary>
-            /// <param name="terms">
-            /// The terms to append to the search query. Each term will be escaped and will be wrapped
-            /// with quotes if it contains whitespaces.
-            /// </param>
+            /// <param name="terms">The terms to append to the search query.</param>
             public void AppendTerms(IReadOnlyList<string> terms)
             {
-                ValidateOrThrow(terms, additionalClauses: terms.Count);
+                ValidateAdditionalClausesOrThrow(terms.Count);
+                ValidateTermsOrThrow(terms);
+
+                AppendSpaceIfNotEmpty();
+
+                for (var i = 0; i < terms.Count; i++)
+                {
+                    if (i > 0)
+                    {
+                        _result.Append(' ');
+                    }
+
+                    AppendEscapedString(terms[i], quoteWhiteSpace: true);
+                }
+            }
+
+            /// <summary>
+            /// Append a clause to boost results that match all terms.
+            /// </summary>
+            /// <param name="terms">All terms that must be matched.</param>
+            /// <param name="boost">The boost for results that match all terms.</param>
+            public void AppendBoostIfMatchAllTerms(IReadOnlyList<string> terms, float boost)
+            {
+                // We will generate a clause for each term and a clause to OR terms together.
+                ValidateAdditionalClausesOrThrow(terms.Count + 1);
+                ValidateTermsOrThrow(terms);
+
+                AppendSpaceIfNotEmpty();
+
+                _result.Append('(');
 
                 for (var i = 0; i < terms.Count; i++)
                 {
-                    if (_result.Length > 0)
+                    if (i > 0)
                     {
                         _result.Append(' ');
                     }
 
+                    _result.Append('+');
                     AppendEscapedString(terms[i], quoteWhiteSpace: true);
                 }
+
+                _result.Append(")^");
+                _result.Append(boost);
             }
 
             /// <summary>
-            /// Append search terms to the query. These terms will be scoped to the specified field. This can be called many times.
+            /// Append a term to the query that is scoped to a specified field.
```

Comment:
Can you explain and leave an XML doc about why we don't just have `AppendScopedTerms` and if someone just has one they pass in `new[] { term }`.

Reply:
Good idea. The API differences are arbitrary and super tailored to the queries we generate, I went ahead and documented that